### ISSUE TITLE
Hold action unreliable

### ISSUE THREAD

**Issue by michalk-k** [home-assistant/iOS#1881]
Model Name: iPhone X
Software Version: iOS15
App version: 2021.8 (2021.216)

**Home Assistant Core Version**
2021.9.7

**Describe the bug**
This issue is not related to this particular iOS nor app nor core versions. It was appearing half year ago, it appears today.
Also not sure it is iOS or Polymer related. I've read other's experience that on Android hold action works as expected.
I can confirm that on PC (win/linux) it works as expected. 

On iOS however attempt to execute hold action is prone to fail. From what I can observe, it seems that it's very sensitive to side-movements of a finger. To the extent that it require super-focusing for a time interval needed to hold action to execute. Otherwise the process is being interrupted. I can succeed for 1 out of 5 attempts only. 
Obviously hold feature in iOS itself (including other apps) works as expected

**To Reproduce**
N/A

**Expected behavior**
More reliable hold event recognition

**Screenshots**
Here is short video with demonstration
https://youtu.be/e-s3B_rzek8
**Additional context**
N/A

**Comment by deluxestyle** [comment]
Having the same issue 
Hold action is not always reacting as it should

**Comment by Nickno95** [comment]
Suggestion to check for hold within the boundary of the object rather than the specific point of initial touch/click? Not sure about how to implement, but if it's using touch events, maybe check if within button on touchend?

**Comment by bgoncal** [comment]
Is this still an issue with latest versions of app and core? Feel free to reopen if needed

**Comment by michalk-k** [comment]
@bgoncal  You closed the issue so quickly...
The reported problem still persist.

**Comment by bgoncal** [comment]
> @bgoncal  You closed the issue so quickly...
> The reported problem still persist.

That's why I mentioned "feel free to reopen", I was doing some clean up in old issues.

Meanwhile, is this behavior also happening when accessing HA over iOS Safari?

**Comment by michalk-k** [comment]
kind of... similar.
But I think I know where the problem is coming from. 

It,s vertical scrolling of main part of ui causes interrupting of any hold action.

I found it recently in vacuum card which provides option to zoom map by two fingers. It's being interrupted as hell. I noticed it happens every-time when the page scrolls (even if I still touch the screen with two fingers)

The same happens to hold action. Simply if your finger moves horizontally a bit, it interrupts hold action.

I'm not sure it's framework or os specific and it can be done anything to improve that. 
I've checked a few apps which implements hold action and vertically scrollable viewports (ie google docs), and seems the reacts on hold better. it's like they disables vertical scroll earlier then waits for longpress time a bit more. Our app doesn't lock vertical scroll at all

with regards

**Comment by bgoncal** [comment]
> kind of... similar. But I think I know where the problem is coming from.
> 
> It,s vertical scrolling of main part of the ui causes interrupting of any hold action.
> 
> I found it recently in vacuum card which provides option to zoom map by two fingers. It's being interrupted as hell. I noticed it happens every-time when the UI page scrolls (even if I still hold the screen with two fingers)
> 
> The same happens to reported hold action. Simply if your finger moves horizontally a bit, it interrupts initial phase of hold action
> 
> I'm not sure it's framework or os specific and it if you can done anything to improve that. I've checked a few apps which implement hold action and vertically scrollable viewports at the same time (ie google docs), and seems they react on hold better. it's like they disable vertical scroll earlier then wait for longpress time condition a bit more. Our app doesn't lock vertical scroll at all
> 
> with regards

Do you experience the same issue if you use it through iOS Safari? (not from within the app)

**Comment by michalk-k** [comment]
Yes. "kind of... similar" was actually the answer to this question.
Any vertical movement, started after hold initialization, cancels the hold action. Seems like the vertical scroll has precedence over the hold action.

**Comment by bgoncal** [comment]
Sorry, I missed your answer, in this case I think you will have more luck bringing this issue to fronend repo, because they are the ones that can fix for browser + apps all at once, fixing in the app would be an workaround that could potentially break again since frontend would be aware of it.
